### DISCUSSION TITLE
CS0016 / CS8034 issues with sample app

### DISCUSSION THREAD

**Discussion by russnash** [CommunityToolkit/Maui#392] [Q&A] [answered]
Please excuse my inexperience, but I have just cloned the CommunityToolkit/Maui repository locally, opened the CommunityToolkit.Maui.Sample.sln solution and attempted to run the sample app on an Android emulator.  However, I am experiencing some CS0016 and CS8034 errors:

`Severity	Code	Description	Project	File	Line	Suppression State	Detail Description
Error	CS0016	Could not write to output file 'C:\Users\russn\source\repos\CommunityToolkit\Maui\samples\CommunityToolkit.Maui.Sample\obj\GeneratedFiles\Microsoft.Maui.Controls.SourceGen\Microsoft.Maui.Controls.SourceGen.CodeBehindGenerator\Pages_Converters_IsStringNotNullOrWhiteSpaceConverterPage.xaml.sg.cs' -- 'Could not find a part of the path 'C:\Users\russn\source\repos\CommunityToolkit\Maui\samples\CommunityToolkit.Maui.Sample\obj\GeneratedFiles\Microsoft.Maui.Controls.SourceGen\Microsoft.Maui.Controls.SourceGen.CodeBehindGenerator\Pages_Converters_IsStringNotNullOrWhiteSpaceConverterPage.xaml.sg.cs'.'	CommunityToolkit.Maui.Sample (net6.0-ios)	C:\Users\russn\source\repos\CommunityToolkit\Maui\samples\CommunityToolkit.Maui.Sample\CSC	1	Active
Error	CS0016	Could not write to output file 'C:\Users\russn\source\repos\CommunityToolkit\Maui\samples\CommunityToolkit.Maui.Sample\obj\GeneratedFiles\CommunityToolkit.Mvvm.SourceGenerators\CommunityToolkit.Mvvm.SourceGenerators.ObservablePropertyGenerator\CommunityToolkit.Maui.Sample.ViewModels.Converters.IsNotNullConverterViewModel.cs' -- 'Could not find a part of the path 'C:\Users\russn\source\repos\CommunityToolkit\Maui\samples\CommunityToolkit.Maui.Sample\obj\GeneratedFiles\CommunityToolkit.Mvvm.SourceGenerators\CommunityToolkit.Mvvm.SourceGenerators.ObservablePropertyGenerator\CommunityToolkit.Maui.Sample.ViewModels.Converters.IsNotNullConverterViewModel.cs'.'	CommunityToolkit.Maui.Sample (net6.0-ios)	C:\Users\russn\source\repos\CommunityToolkit\Maui\samples\CommunityToolkit.Maui.Sample\CSC	1	Active
Error	CS0016	Could not write to output file 'C:\Users\russn\source\repos\CommunityToolkit\Maui\samples\CommunityToolkit.Maui.Sample\obj\GeneratedFiles\CommunityToolkit.Mvvm.SourceGenerators\CommunityToolkit.Mvvm.SourceGenerators.ObservablePropertyGenerator\CommunityToolkit.Maui.Sample.ViewModels.Converters.IsNullConverterViewModel.cs' -- 'Could not find a part of the path 'C:\Users\russn\source\repos\CommunityToolkit\Maui\samples\CommunityToolkit.Maui.Sample\obj\GeneratedFiles\CommunityToolkit.Mvvm.SourceGenerators\CommunityToolkit.Mvvm.SourceGenerators.ObservablePropertyGenerator\CommunityToolkit.Maui.Sample.ViewModels.Converters.IsNullConverterViewModel.cs'.'	CommunityToolkit.Maui.Sample (net6.0-ios)	C:\Users\russn\source\repos\CommunityToolkit\Maui\samples\CommunityToolkit.Maui.Sample\CSC	1	Active
Error	CS0016	Could not write to output file 'C:\Users\russn\source\repos\CommunityToolkit\Maui\samples\CommunityToolkit.Maui.Sample\obj\GeneratedFiles\CommunityToolkit.Mvvm.SourceGenerators\CommunityToolkit.Mvvm.SourceGenerators.ObservablePropertyGenerator\CommunityToolkit.Maui.Sample.ViewModels.Converters.StateToBooleanConverterViewModel.cs' -- 'Could not find a part of the path 'C:\Users\russn\source\repos\CommunityToolkit\Maui\samples\CommunityToolkit.Maui.Sample\obj\GeneratedFiles\CommunityToolkit.Mvvm.SourceGenerators\CommunityToolkit.Mvvm.SourceGenerators.ObservablePropertyGenerator\CommunityToolkit.Maui.Sample.ViewModels.Converters.StateToBooleanConverterViewModel.cs'.'	CommunityToolkit.Maui.Sample (net6.0-ios)	C:\Users\russn\source\repos\CommunityToolkit\Maui\samples\CommunityToolkit.Maui.Sample\CSC	1	Active
Error	CS0016	Could not write to output file 'C:\Users\russn\source\repos\CommunityToolkit\Maui\samples\CommunityToolkit.Maui.Sample\obj\GeneratedFiles\CommunityToolkit.Mvvm.SourceGenerators\CommunityToolkit.Mvvm.SourceGenerators.ObservablePropertyGenerator\CommunityToolkit.Maui.Sample.ViewModels.Views.DrawingViewViewModel.cs' -- 'Could not find a part of the path 'C:\Users\russn\source\repos\CommunityToolkit\Maui\samples\CommunityToolkit.Maui.Sample\obj\GeneratedFiles\CommunityToolkit.Mvvm.SourceGenerators\CommunityToolkit.Mvvm.SourceGenerators.ObservablePropertyGenerator\CommunityToolkit.Maui.Sample.ViewModels.Views.DrawingViewViewModel.cs'.'	CommunityToolkit.Maui.Sample (net6.0-ios)	C:\Users\russn\source\repos\CommunityToolkit\Maui\samples\CommunityToolkit.Maui.Sample\CSC	1	Active
Error	CS0016	Could not write to output file 'C:\Users\russn\source\repos\CommunityToolkit\Maui\samples\CommunityToolkit.Maui.Sample\obj\GeneratedFiles\CommunityToolkit.Maui.SourceGenerators\CommunityToolkit.Maui.SourceGenerators.Generators.TextColorToGenerator\ButtonTextColorTo_89dd3848-cfd6-4e57-a910-f2cf95913ab8.g.shared.cs' -- 'Could not find a part of the path 'C:\Users\russn\source\repos\CommunityToolkit\Maui\samples\CommunityToolkit.Maui.Sample\obj\GeneratedFiles\CommunityToolkit.Maui.SourceGenerators\CommunityToolkit.Maui.SourceGenerators.Generators.TextColorToGenerator\ButtonTextColorTo_89dd3848-cfd6-4e57-a910-f2cf95913ab8.g.shared.cs'.'	CommunityToolkit.Maui.Sample (net6.0-ios)	C:\Users\russn\source\repos\CommunityToolkit\Maui\samples\CommunityToolkit.Maui.Sample\CSC	1	Active
Error	CS0016	Could not write to output file 'C:\Users\russn\source\repos\CommunityToolkit\Maui\samples\CommunityToolkit.Maui.Sample\obj\GeneratedFiles\CommunityToolkit.Maui.SourceGenerators\CommunityToolkit.Maui.SourceGenerators.Generators.TextColorToGenerator\DatePickerTextColorTo_cc03e134-6126-4ab6-89bb-bf2c73e45b08.g.shared.cs' -- 'Could not find a part of the path 'C:\Users\russn\source\repos\CommunityToolkit\Maui\samples\CommunityToolkit.Maui.Sample\obj\GeneratedFiles\CommunityToolkit.Maui.SourceGenerators\CommunityToolkit.Maui.SourceGenerators.Generators.TextColorToGenerator\DatePickerTextColorTo_cc03e134-6126-4ab6-89bb-bf2c73e45b08.g.shared.cs'.'	CommunityToolkit.Maui.Sample (net6.0-ios)	C:\Users\russn\source\repos\CommunityToolkit\Maui\samples\CommunityToolkit.Maui.Sample\CSC	1	Active
Error	CS0016	Could not write to output file 'C:\Users\russn\source\repos\CommunityToolkit\Maui\samples\CommunityToolkit.Maui.Sample\obj\GeneratedFiles\CommunityToolkit.Maui.SourceGenerators\CommunityToolkit.Maui.SourceGenerators.Generators.TextColorToGenerator\EditorTextColorTo_dbc81c53-88dd-4546-9cb3-cf743d17d2ae.g.shared.cs' -- 'Could not find a part of the path 'C:\Users\russn\source\repos\CommunityToolkit\Maui\samples\CommunityToolkit.Maui.Sample\obj\GeneratedFiles\CommunityToolkit.Maui.SourceGenerators\CommunityToolkit.Maui.SourceGenerators.Generators.TextColorToGenerator\EditorTextColorTo_dbc81c53-88dd-4546-9cb3-cf743d17d2ae.g.shared.cs'.'	CommunityToolkit.Maui.Sample (net6.0-ios)	C:\Users\russn\source\repos\CommunityToolkit\Maui\samples\CommunityToolkit.Maui.Sample\CSC	1	Active
Error	CS0016	Could not write to output file 'C:\Users\russn\source\repos\CommunityToolkit\Maui\samples\CommunityToolkit.Maui.Sample\obj\GeneratedFiles\CommunityToolkit.Maui.SourceGenerators\CommunityToolkit.Maui.SourceGenerators.Generators.TextColorToGenerator\EntryTextColorTo_1eedfe5e-d3f3-4b59-96f1-78a978ea0c84.g.shared.cs' -- 'Could not find a part of the path 'C:\Users\russn\source\repos\CommunityToolkit\Maui\samples\CommunityToolkit.Maui.Sample\obj\GeneratedFiles\CommunityToolkit.Maui.SourceGenerators\CommunityToolkit.Maui.SourceGenerators.Generators.TextColorToGenerator\EntryTextColorTo_1eedfe5e-d3f3-4b59-96f1-78a978ea0c84.g.shared.cs'.'	CommunityToolkit.Maui.Sample (net6.0-ios)	C:\Users\russn\source\repos\CommunityToolkit\Maui\samples\CommunityToolkit.Maui.Sample\CSC	1	Active
Error	CS0016	Could not write to output file 'C:\Users\russn\source\repos\CommunityToolkit\Maui\samples\CommunityToolkit.Maui.Sample\obj\GeneratedFiles\CommunityToolkit.Maui.SourceGenerators\CommunityToolkit.Maui.SourceGenerators.Generators.TextColorToGenerator\LabelTextColorTo_14ccb1f0-5807-43b1-b879-62757e2bbc55.g.shared.cs' -- 'Could not find a part of the path 'C:\Users\russn\source\repos\CommunityToolkit\Maui\samples\CommunityToolkit.Maui.Sample\obj\GeneratedFiles\CommunityToolkit.Maui.SourceGenerators\CommunityToolkit.Maui.SourceGenerators.Generators.TextColorToGenerator\LabelTextColorTo_14ccb1f0-5807-43b1-b879-62757e2bbc55.g.shared.cs'.'	CommunityToolkit.Maui.Sample (net6.0-ios)	C:\Users\russn\source\repos\CommunityToolkit\Maui\samples\CommunityToolkit.Maui.Sample\CSC	1	Active
Error	CS0016	Could not write to output file 'C:\Users\russn\source\repos\CommunityToolkit\Maui\samples\CommunityToolkit.Maui.Sample\obj\GeneratedFiles\CommunityToolkit.Maui.SourceGenerators\CommunityToolkit.Maui.SourceGenerators.Generators.TextColorToGenerator\PickerTextColorTo_01cf57cf-a1aa-4d70-981e-59c2d0d7801a.g.shared.cs' -- 'Could not find a part of the path 'C:\Users\russn\source\repos\CommunityToolkit\Maui\samples\CommunityToolkit.Maui.Sample\obj\GeneratedFiles\CommunityToolkit.Maui.SourceGenerators\CommunityToolkit.Maui.SourceGenerators.Generators.TextColorToGenerator\PickerTextColorTo_01cf57cf-a1aa-4d70-981e-59c2d0d7801a.g.shared.cs'.'	CommunityToolkit.Maui.Sample (net6.0-ios)	C:\Users\russn\source\repos\CommunityToolkit\Maui\samples\CommunityToolkit.Maui.Sample\CSC	1	Active
Error	CS0016	Could not write to output file 'C:\Users\russn\source\repos\CommunityToolkit\Maui\samples\CommunityToolkit.Maui.Sample\obj\GeneratedFiles\CommunityToolkit.Maui.SourceGenerators\CommunityToolkit.Maui.SourceGenerators.Generators.TextColorToGenerator\RadioButtonTextColorTo_85c75634-8102-4a3c-95f4-c3f328cb75c7.g.shared.cs' -- 'Could not find a part of the path 'C:\Users\russn\source\repos\CommunityToolkit\Maui\samples\CommunityToolkit.Maui.Sample\obj\GeneratedFiles\CommunityToolkit.Maui.SourceGenerators\CommunityToolkit.Maui.SourceGenerators.Generators.TextColorToGenerator\RadioButtonTextColorTo_85c75634-8102-4a3c-95f4-c3f328cb75c7.g.shared.cs'.'	CommunityToolkit.Maui.Sample (net6.0-ios)	C:\Users\russn\source\repos\CommunityToolkit\Maui\samples\CommunityToolkit.Maui.Sample\CSC	1	Active
Error	CS0016	Could not write to output file 'C:\Users\russn\source\repos\CommunityToolkit\Maui\samples\CommunityToolkit.Maui.Sample\obj\GeneratedFiles\CommunityToolkit.Maui.SourceGenerators\CommunityToolkit.Maui.SourceGenerators.Generators.TextColorToGenerator\SearchBarTextColorTo_08597350-e423-4585-a2a1-cba3220b8057.g.shared.cs' -- 'Could not find a part of the path 'C:\Users\russn\source\repos\CommunityToolkit\Maui\samples\CommunityToolkit.Maui.Sample\obj\GeneratedFiles\CommunityToolkit.Maui.SourceGenerators\CommunityToolkit.Maui.SourceGenerators.Generators.TextColorToGenerator\SearchBarTextColorTo_08597350-e423-4585-a2a1-cba3220b8057.g.shared.cs'.'	CommunityToolkit.Maui.Sample (net6.0-ios)	C:\Users\russn\source\repos\CommunityToolkit\Maui\samples\CommunityToolkit.Maui.Sample\CSC	1	Active
Error	CS0016	Could not write to output file 'C:\Users\russn\source\repos\CommunityToolkit\Maui\samples\CommunityToolkit.Maui.Sample\obj\GeneratedFiles\CommunityToolkit.Maui.SourceGenerators\CommunityToolkit.Maui.SourceGenerators.Generators.TextColorToGenerator\TimePickerTextColorTo_0ab1f126-2bfc-424c-b1a2-9eea6a284a93.g.shared.cs' -- 'Could not find a part of the path 'C:\Users\russn\source\repos\CommunityToolkit\Maui\samples\CommunityToolkit.Maui.Sample\obj\GeneratedFiles\CommunityToolkit.Maui.SourceGenerators\CommunityToolkit.Maui.SourceGenerators.Generators.TextColorToGenerator\TimePickerTextColorTo_0ab1f126-2bfc-424c-b1a2-9eea6a284a93.g.shared.cs'.'	CommunityToolkit.Maui.Sample (net6.0-ios)	C:\Users\russn\source\repos\CommunityToolkit\Maui\samples\CommunityToolkit.Maui.Sample\CSC	1	Active
Warning	CS8034	Unable to load Analyzer assembly C:\Users\russn\source\repos\CommunityToolkit\Maui\src\CommunityToolkit.Maui.SourceGenerators\bin\Debug\netstandard2.0\CommunityToolkit.Maui.SourceGenerators.dll: Could not find file 'C:\Users\russn\source\repos\CommunityToolkit\Maui\src\CommunityToolkit.Maui.SourceGenerators\bin\Debug\netstandard2.0\CommunityToolkit.Maui.SourceGenerators.dll'.	CommunityToolkit.Maui.Sample (net6.0-ios)		1	Active	System.IO.FileNotFoundException: Could not find file 'C:\Users\russn\source\repos\CommunityToolkit\Maui\src\CommunityToolkit.Maui.SourceGenerators\bin\Debug\netstandard2.0\CommunityToolkit.Maui.SourceGenerators.dll'.
File name: 'C:\Users\russn\source\repos\CommunityToolkit\Maui\src\CommunityToolkit.Maui.SourceGenerators\bin\Debug\netstandard2.0\CommunityToolkit.Maui.SourceGenerators.dll'
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share)
   at Roslyn.Utilities.StandardFileSystem.OpenFile(String filePath, FileMode mode, FileAccess access, FileShare share)
   at Roslyn.Utilities.CommonCompilerFileSystemExtensions.OpenFileWithNormalizedException(ICommonCompilerFileSystem fileSystem, String filePath, FileMode fileMode, FileAccess fileAccess, FileShare fileShare)
   at Microsoft.CodeAnalysis.ModuleMetadata.CreateFromFile(String path)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerFileReference.GetAnalyzerTypeNameMap(String fullPath, Type attributeType, AttributeLanguagesFunc languagesFunc)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerFileReference.Extensions`1.GetExtensionTypeNameMap()
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerFileReference.Extensions`1.AddExtensions(Builder builder, String language, Func`2 shouldInclude)
-----

`

I've checked the files reported as "Could not write to output file" in some of the CS0016 errors, and in each case tested the file does not exist although the path is valid.  The CS8034 error at the end reports that it "Could not find file..." for the listed CommunityToolkit.Maui.SourceGenerators.dll, however it does exist in the file system.

I'm sure I am doing something wrong somewhere.

Many thanks in advance for any help!

Russ.

**Comment by bijington** [comment, marked as answer]
@russnash no need to apologise. This issue is sadly a common one that occurs on Windows because by default it has a limit on the length of the path allowed.

You can disable the path length limit by following the steps in this article: https://docs.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=cmd#enable-long-paths-in-windows-10-version-1607-and-later

**Reply by pictos** [reply]
A second workaround would be to comment [these lines](https://github.com/CommunityToolkit/Maui/blob/main/samples/CommunityToolkit.Maui.Sample/CommunityToolkit.Maui.Sample.csproj#L23-L24)

**Reply by russnash** [reply]
That makes sense, thank you guys!

**Comment by CliffAgius** [comment]
Closed as answered
